src/plugins/intel_gna/src/transformations/aszp_decomposition.cpp

```diff
@@ -0,0 +1,231 @@
+
+	auto convolution_children = conv->output(0).get_target_inputs();
+	auto convolution_bias = std::dynamic_pointer_cast<ngraph::opset8::Add>(convolution_children.begin()->get_node()->shared_from_this());
+	
+	std::shared_ptr<ngraph::opset8::Transpose> transpose_after;
+	
+	if (nullptr != convolution_bias) {
+		auto add_children = convolution_bias->output(0).get_target_inputs();
+        if (add_children.size() != 1) return nullptr;
+        
+		transpose_after = std::dynamic_pointer_cast<ngraph::opset8::Transpose>(add_children.begin()->get_node()->shared_from_this());
+	} else transpose_after = std::dynamic_pointer_cast<ngraph::opset8::Transpose>(convolution_children.begin()->get_node()->shared_from_this());
+	
+	if (transpose_after == nullptr) return nullptr;
+
+	return transpose_before;
+}
+
+
+
+static std::tuple<uint64_t, uint64_t, uint64_t, uint64_t> GetInputDimensions(ov::Shape input_shape) {   
+	uint64_t N = input_shape[0]; // batchsize
+	uint64_t H = input_shape[1]; // no input channels (input depth)
+	uint64_t W = input_shape[2]; // input height
+	uint64_t C = input_shape[3]; // input widith
+
+	return std::make_tuple(N,H,W,C);
+}
+
+
+
+std::shared_ptr<ov::op::v0::Concat> ConcatZeros(
+	uint32_t pad_begin,
+	uint32_t pad_end,
+	std::shared_ptr<ov::Node> padding_const,
+	std::shared_ptr<ov::Node> input_node) {
+
+	OutputVector concat_vector;
+	if (pad_begin > pad_end) {
+		concat_vector.push_back(padding_const->output(0));
+		concat_vector.push_back(input_node->output(0));
+	}else {
+		concat_vector.push_back(input_node->output(0));
+		concat_vector.push_back(padding_const->output(0));	
+	}
+	return std::make_shared<ngraph::op::Concat>(concat_vector,1);
+
+}
+
+
+
+
+Output<Node> DecomposeHeight(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, C, H, W;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end);
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Hpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{H,W*C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+
+		auto padding_const = CreateZeroConst(Shape{W*C,Hpad});
+		auto new_concat    = ConcatZeros(Hbegin,Hend,padding_const,new_transpose);
+		
+		auto new_untranspose = CreateTranspose(new_concat->output(0));	// Shape {W*C,H+Hpad}
+
+		if (0 == Wpad) return CreateReshape(new_untranspose->output(0), 4, Shape{N, H+Hpad, W ,C})->output(0);
+		else return (new_untranspose->output(0));
+	}else return input;
+
+
+}
+
+
+
+Output<Node> DecomposeWidth(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, H, W, C;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end); //todo cleanup. maybe get only what we need?
```

Comment:
please remove unused elements.

Reply:
I separated the extract_padding function into separate ones for height and width. The could be merged into one function with flag argument specifying if it's for width and height but I avoided that for better readability. If the flag approach would be better here please let me know